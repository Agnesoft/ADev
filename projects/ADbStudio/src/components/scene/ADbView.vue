<template>
    <div class="adb-view" :class="{'search-mode': searchedWord !== ''}">
        <div class="view">
            <Edge v-for="(edges,key) in edgesGrouped" :key="key" :elements="edges" :viewSize="viewSize" :class="{selected: edgeIsSelected(edges)}" v-on="$listeners" />
            <Node v-for="node in nodes" :key="node.id" :element="node" :viewSize="viewSize" :class="{selected: node.id == searchedWord}" v-on="$listeners" />
        </div>
    </div>
</template>

<script>
import Node from "./Node";
import Edge from "./Edge";
import { mapGetters } from 'vuex'

export default {
    name: "ADbView",
    components: {
        Node,
        Edge,
    },
    data(){
        return {
            viewWidth: 0,
            viewHeight: 0,
        }
    },
    props: {
        searchedWord: String,
    },
    computed: {
        ...mapGetters({
            nodes: 'scene/getNodes',
            edges: 'scene/getEdges',
            edgesGrouped: 'scene/getEdgesGrouped',
        }),
        viewSize(){
            return {
                width: this.viewWidth,
                height: this.viewHeight
            }
        }
    },
    methods: {
        onResize() {
            this.viewWidth = document.documentElement.clientWidth;
            this.viewHeight = document.documentElement.clientHeight;
        },
        edgeIsSelected(edges){
            return edges.find((edge) => edge.id == this.searchedWord) ? true : false;
        }
    },
    mounted() {
        window.addEventListener('resize', this.onResize)
        this.onResize();
    },

    beforeDestroy() {
        window.removeEventListener('onmousewheel', this.onResize)
    }

}
</script>

<style scoped>
    .adb-view{
        position: relative;
        transform: scale(var(--scale,1)) translate(var(--move-x,0), var(--move-y,0));
        width: 100%;
        height: 100%;
    }
</style>
