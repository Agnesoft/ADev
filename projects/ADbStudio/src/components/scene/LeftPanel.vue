<template>

    <SlideXTransition :opened="has_data"  >
        <div class="left-panel">
            <div class="tabs">
                <div class="tab-button" :class="{active: tab.key==active_tab_key}" v-for="tab in tabs" :key="tab.key" :title="tab.title" >
                    <BaseButton class="btn-circle" :class="{'btn-primary' : tab.key==active_tab_key,'btn-light' : tab.key!=active_tab_key}" @click="buttonClicked(tab.key)">
                        <BaseIcon :icon_key="tab.key" />
                    </BaseButton>
                </div>
            </div>
            <SlideXTransition :opened="active_tab_key != -1"  >
                <div class="tab-box">
                    <header class="tab-header">
                        {{tab_title}}
                    </header>
                    <transition name="fade" mode="out-in">
                        <component class="tab-content" v-bind:is="component_content"  v-on="$listeners"></component>
                    </transition>
                </div>
            </SlideXTransition>    
        </div>
    </SlideXTransition> 
</template>

<script>
import SlideXTransition from "../transitions/SlideXTransition";
import { mapGetters } from 'vuex'

export default {
    name: "LeftPanel",
    components: {
        SlideXTransition,
        NodeListData: () => import('./NodeListData'),
        EdgeListData: () => import('./EdgeListData'),
        RawData: () => import('./RawData')
    },
    data() {
        return {
            active_tab_key: -1,
            tabs: [
                {
                    key: 'nodes',
                    title: this.$t('list-of-nodes'),
                    component_content: "NodeListData",
                },
                {
                    key: 'edges',
                    title: this.$t('list-of-edges'),
                    component_content: "EdgeListData",
                },
                {
                    key: 'data',
                    title: this.$t('raw-data'),
                    component_content: "RawData",
                },
            ]
        }
    },
    computed: {
        ...mapGetters({
            data: 'scene/getData',
        }),
        active_tab(){
            if(!this.active_tab_key){
                return false;
            }
            let active_tab_key = this.active_tab_key;
            let tab = this.tabs.find((elem) => elem.key == active_tab_key);
            if(!tab){
                return false;
            }
            return tab;
        },
        component_content(){
            let active_tab_key = this.active_tab_key;
            if(active_tab_key){
                return this.active_tab.component_content;
            }
            return false;
        },
        tab_title(){
            let active_tab_key = this.active_tab_key;
            if(active_tab_key){
                return this.active_tab.title;
            }
            return '';
        },
        has_data(){
            return !(Object.keys(this.data).length === 0 && this.data.constructor === Object);
        }
    },
    methods: {
        buttonClicked(tab_key){
            let tab = this.tabs.find((elem) => elem.key == tab_key);
            if(!tab){
                return;
            }
            if(this.active_tab_key === tab_key){
                this.active_tab_key = -1;
            } else {
                this.active_tab_key = tab_key;
            }
        },
    }
}
</script>

<style scoped>
    .left-panel{
        display: flex;
    }
    .tabs{
        display: flex;
        flex-direction: column;
        margin-right: var(--grid-gap);
    }
    .btn{
        margin: 0.2rem;
        transition: all 0.2s ease;
    }
    .tab-box{
        border-radius: 0.4rem;
        overflow: hidden;
        background: white;
        display: flex;
        flex-direction: column;
        max-height: 80vh;
        --shadow: var(--dark-color) 1px 1px 5px 0px;
        -webkit-box-shadow:  var(--shadow);
        -moz-box-shadow:  var(--shadow);
        box-shadow:  var(--shadow);
    }
    .tab-header{
        padding: 0.5rem;
        white-space: nowrap;
    }
    .tab-content{
        width: 15rem;
        padding: 0.5rem;
        margin: 0;
        height: 100%;
        overflow: auto;
    }
    .fade-enter-active {
        transition: all 0.3s
    }
    .fade-enter {
        opacity: 0;
    }
    .fade-leave-active {
        transition: all 0.3s;
        opacity: 0;
    }
    .tab-header{
        background: var(--primary-color);
        color: var(--light-color);
    }
</style>
