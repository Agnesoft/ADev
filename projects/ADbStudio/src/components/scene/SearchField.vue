<template>
    <div class="search-field" :class="{opened: showContent}">
        <SlideXTransition :opened="showContent">
            <form class="search-form" @submit.prevent="onSubmit">
                <input type="text" ref="search_input" class="search-input" v-model.trim="search" :placeholder="$t('search-element')"/>
            </form>
        </SlideXTransition>
        <BaseButton class="btn-light" @click="handleClick">
            <svg width="100%" height="100%">
                <use xlink:href="../../assets/images/icons.svg#search" x = 0 y = 0></use>
            </svg>
        </BaseButton>
    </div>
</template>

<script>
import SlideXTransition from "@/components/transitions/SlideXTransition";
export default {
    name: "SearchField",
    components: {
        SlideXTransition,
    },
    data(){
        return {
            search: '',
            showContent: false,
        }
    },
    methods: {
        handleClick(){
            if(this.search != ''){
                this.onSubmit();
                return;
            } 
            this.showContent = !this.showContent; 
            if(this.showContent){
                let component = this;
                this.$nextTick()
                .then(function () {
                    component.$refs.search_input.focus();
                })
            }
        },
        // handleFocusOut(){
        //     this.showContent = false;
        // },
        onSubmit(){
            this.$emit('search',this.search);
            this.search = ''
        }
    },
}
</script>

<style scoped>
    .search-field{
        display: flex;
        height: var(--button-size);
        border-radius: calc(var(--button-size) * 0.5);
        transition: border var(--transition-ease);
    }
    .search-field.opened{
        border: 1px solid var(--light-primary-color);
    }
    .btn-light{
        width: var(--button-size);
        height: var(--button-size);
        padding: 0.2rem;
    }
    .base-expandable{
        z-index: 1;
        margin: 0.2rem;
        /* --shadow: var(--primary-color) 0px 0px 2px 2px;
        -webkit-box-shadow:  var(--shadow);
        -moz-box-shadow:  var(--shadow);
        box-shadow:  var(--shadow); */
        display: inline-block;
    }
    .search-input{
        border: none;
        padding-left: 1rem;
        background: none;
        height: 100%;
    }
</style>
