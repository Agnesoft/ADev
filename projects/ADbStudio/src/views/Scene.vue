<template>
    <div class="scene">
        <form class="command-form" @submit.prevent="sendCommand">
            <textarea class="command" v-model.trim="command" :placeholder="$t('write-command')" rows="1" />
            <input type="submit" :value="$t('run')" class="submit-button btn btn-primary">
        </form>
        <LeftPanel @showData="showData" />
        <div class="view-container" @wheel.prevent="onMouseWheel" @dragover.prevent @drop="onDrop" @dragstart="onDragStart">
            <ADbView id="adb-view" ref="adb_view" :zoom="zoom" draggable="true" :style="viewStyle" :searchedWord="searchedWord"  @showData="showData"/>
            <div class="search-wrap">    
                <SearchField @search="search"/>          
                <BaseMessage class="error" :opened="error === 'search'" @close="error=''">
                    {{ $t("element-not-found") }}
                </BaseMessage> 
                <BaseMessage class="info" :opened="searchedWord != ''" @close="resetView">
                    {{ $t('searched-element-word',{word: searchedWord}) }}
                </BaseMessage>
            </div>
        </div>
        <DataWrapper :show="displayData" :elements="elementsData" :dataStyle="elementDataStyle" @exploreData="exploreData" />
        <RightPanel />
        <div class="controls">
            <BaseButton class="btn-default reset" @click="resetView">{{$t('reset-view')}}</BaseButton>
            <BaseButton class="btn-default plus" @click="zoomPlus">+</BaseButton>
            <BaseButton class="btn-default minus" @click="zoomMinus">-</BaseButton>
        </div>
    </div>
</template>

<script>
import ADbView from '@/components/scene/ADbView.vue';
import LeftPanel from '@/components/scene/LeftPanel.vue';
import SearchField from '@/components/scene/SearchField.vue';
import DataWrapper from "@/components/scene/DataWrapper";
import RightPanel from "@/components/scene/RightPanel";

import { mapActions,mapGetters } from 'vuex';

export default {
    name: "Scene",
    components: {
        SearchField,
        ADbView,
        DataWrapper,
        LeftPanel,
        RightPanel
    },
    data(){
        return {
            command: '',
            zoom: 0,
            move: {
                x: 0,
                y: 0
            },
            startDragCoordinates: {
                x: 0,
                y: 0
            },
            searchedWord: '',
            error: '',
            dataShow: false,
            dataExporation: false,
            elementsData: [],
            dataPos: {
                x: 0,
                y: 0
            },
        }
    },
    methods: {
        ...mapActions({
            send_command: 'scene/sendCommand',
        }),
        sendCommand(){
            if(this.command != ""){
                this.send_command({command: this.command});
                this.command = '';
                this.resetView();
            }
        },
        onMouseWheel(event){
            event.deltaY < 0 ? this.zoomPlus() : this.zoomMinus(); 
        },
        onDragStart(event){
            this.startDragCoordinates.x = event.clientX;
            this.startDragCoordinates.y = event.clientY;
        },
        onDrop(event){
            this.move.x += (event.clientX - this.startDragCoordinates.x)/this.scale;
            this.move.y += (event.clientY - this.startDragCoordinates.y)/this.scale;
        },
        resetView(){
            this.move.x = 0;
            this.move.y = 0;
            this.zoom = 0;
            this.startDragCoordinates.x = 0;
            this.startDragCoordinates.y = 0;
            this.resetSearch();
        },
        resetSearch(){
            this.searchedWord = '';
            this.error = '';
        },
        zoomPlus(){
            this.zoom++;
        },
        zoomMinus(){
            this.zoom--;
            this.zoom = this.zoom < -9 ? -9 : this.zoom;
        },
        search(word){
            this.resetView();
            
            this.error = "search";
            this.searchedWord = '';
            if(isNaN(word) || word == 0){
                return false;
            }
            if(word < 0){
                let edge = this.edges.find((edge) => edge.id == word);
                if(edge){
                    this.searchedWord = word;
                    this.error = '';
                }
            } else if(word > 0) {
                let node = this.nodes.find((node) => node.id == word);
                if(node){
                    this.searchedWord = word;
                    this.error = '';
                }
            } 
        },
        showData(elements,event) {
            //this.elementsData = [];
            if(elements === false || elements.length === 0){
                setTimeout(() => this.dataShow = false, 1000);
            } else {
                this.elementsData = elements;
                // if(Object.prototype.hasOwnProperty.call(element, 'data')){
                //     this.data = element.data;
                // }
                this.dataShow = true;
                this.dataPos.y = event.clientY - this.$el.getBoundingClientRect().top + 15;
                this.dataPos.x = event.clientX - this.$el.getBoundingClientRect().left + 15;
            }
        },
        exploreData(explore){
            this.dataExporation = explore;
        }
    },
    computed: {
        ...mapGetters({
            nodes: 'scene/getNodes',
            edges: 'scene/getEdges',
        }),
        viewStyle: function(){
            return {
                "--move-x": this.move.x.toString()+"px",
                "--move-y": this.move.y.toString()+"px", 
                '--scale': this.scale,
            }
        },
        scale: function(){
            let scale = 1;
            scale = 1 + this.zoom/10;
            scale = scale < 0.1 ? 0.1 : scale;
            return scale;
        },
        elementDataStyle(){
            return {
                top: this.dataPos.y+"px",
                left: this.dataPos.x+"px"
            }
        },
        displayData(){
            return this.dataShow || this.dataExporation;
        }

    }
}
</script>

<style scoped>
    .scene{
        position: relative;
        display: grid;
        grid-template-columns: max-content 1fr max-content;
        grid-template-rows: max-content minmax(0, 1fr) max-content;
        grid-template-areas: 
            'command command command'
            'leftpanel view rightpanel'
            'leftpanel controls rightpanel'
            ;
        grid-gap: 0.4rem;
    }
    .command-form{
        grid-area: command;
        display: flex;
        max-width: 80rem;
    }
    .command{
        flex-grow: 2;
        resize: vertical;
        margin-right: 0.2rem;
        border: 1px solid #DDD;
        font: var(--primary-font);
        /* max-width: 50rem */
    }
    .view-container{
        grid-area: view;
        overflow: hidden;
        position: relative;
        background: white;
        box-shadow: inset  0 0 5px var(--light-color);
    }
    #adb-view{
        position: absolute;
    }
    .controls{
        grid-area: controls;
    }
    .controls>.btn{
        margin-right: 0.2rem;
    }
    .left-panel{
        grid-area: leftpanel;
    }
    
    .search-wrap{
        position: absolute;
        right: 0;
        top: 0;
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: max-content max-content;
        grid-template-areas: 
            'input'
            'message'
            ;
        grid-gap: 0.4rem;
        /* flex-direction: column; */
        padding: 0.4rem;
        /* align-items: flex-end; */
    }
    .search-field{
        grid-area: input;
        place-self: center right;
    }
    .base-message{
        grid-area: message;
    }
    .data-wrapper{
        position: absolute;
    }
    .right-panel{
        grid-area: rightpanel;
    }
</style>
