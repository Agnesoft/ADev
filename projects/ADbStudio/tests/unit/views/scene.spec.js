import { mount  } from '@vue/test-utils';
import Vuex from 'vuex';
import Scene from '@/views/Scene.vue';
import SearchField from "@/components/scene/SearchField";
import ADbView from "@/components/scene/ADbView";
import LeftPanel from "@/components/scene/LeftPanel";
import scene_store from '../../../src/store/modules/scene'
//import NodeInfo from "@/components/scene/NodeInfo";

import { localVue, i18n,scene_state } from '../TestUtils'


describe('Scene',() => {
    let wrapper;
    let actions;
    let store;
    let fadeTransitionStub; 

    beforeEach(() => {
        fadeTransitionStub = () => ({
            props: {
                opened: Boolean,
            },
            template: '<div v-if="opened" ><slot></slot></div>',
          });
        
        localVue.use(Vuex)
        actions = {
            sendCommand: jest.fn(),
        }
        store = new Vuex.Store({
            modules: {
                scene: {
                    namespaced: true,
                    actions,
                    state: scene_state,
                    getters: scene_store.getters
                }
              }
        });
        
        wrapper = mount(Scene, {
            localVue,
            i18n,
            store,
            stubs: {
                BaseMessage:  fadeTransitionStub(),
            }
        });
    })
    it('renders all elements', () => {
        expect(wrapper.find(".command").exists()).toBe(true);
        expect(wrapper.findComponent(SearchField).exists()).toBe(true);
        expect(wrapper.findComponent(ADbView).exists()).toBe(true);
        expect(wrapper.findComponent(LeftPanel).exists()).toBe(true);
        expect(wrapper.find(".reset").exists()).toBe(true);
        expect(wrapper.find(".plus").exists()).toBe(true);
        expect(wrapper.find(".minus").exists()).toBe(true);
    })
    it('sends command, fetches data and clear the input', async () => {
        wrapper.setData({command: 'test command'});
        await wrapper.find('form.command-form').trigger('submit.prevent');
        expect(wrapper.find(".command").element.value).toBe("");
        expect(actions.sendCommand).toHaveBeenCalled();
    })
    it('resets the scene after clicking reset button',async () => {
        wrapper.vm.move.x = -1;
        wrapper.vm.move.y = -1;
        wrapper.vm.zoom = -1;
        wrapper.vm.startDragCoordinates.x = -1;
        wrapper.vm.startDragCoordinates.y = -1;
        wrapper.vm.searchedWord = '-1';
        wrapper.vm.error = '-1';

        await wrapper.find('.reset').trigger('click');
        expect(wrapper.vm.move.x).toBe(0);
        expect(wrapper.vm.move.y).toBe(0);
        expect(wrapper.vm.zoom).toBe(0);
        expect(wrapper.vm.startDragCoordinates.x).toBe(0);
        expect(wrapper.vm.startDragCoordinates.x).toBe(0);
        expect(wrapper.vm.searchedWord).toBe('');
        expect(wrapper.vm.error).toBe('');
    })
    it('increases zoom after clicking plus button',async () => {
        await wrapper.find('.plus').trigger('click');
        expect(wrapper.vm.zoom).toBe(1);
    });
    it('decreases zoom after clicking minus button',async () => {
        await wrapper.find('.minus').trigger('click');
        expect(wrapper.vm.zoom).toBe(-1);
    });
    it('displays error message after searching invalid element', async () => {
        expect(wrapper.find(".search-wrap .error").exists()).toBe(false);
        await wrapper.findComponent(SearchField).vm.$emit('search','fdad');
        expect(wrapper.find(".search-wrap .error").exists()).toBe(true);
    });
    it('displays search result after searching ivalid element', async () => {
        expect(wrapper.find(".search-wrap .info").exists()).toBe(false);
        await wrapper.findComponent(SearchField).vm.$emit('search','1');
        expect(wrapper.find(".search-wrap .info").exists()).toBe(true);
        expect(wrapper.find(".search-mode").exists()).toBe(true);
        expect(wrapper.find(".node.selected").exists()).toBe(true);

    });
})
