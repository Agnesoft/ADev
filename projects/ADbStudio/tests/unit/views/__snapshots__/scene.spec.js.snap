// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scene renders correctly (snapshot) 1`] = `
<div
  class="scene"
>
  <form
    class="command-form"
  >
    <textarea
      class="command"
      placeholder="Write command"
      rows="1"
    />
     
    <input
      class="submit-button btn btn-primary"
      type="submit"
      value="Execute"
    />
  </form>
   
  <transition-stub
    name="slide-x"
  >
    <div
      class="left-panel"
    >
      <div
        class="tabs"
      >
        <div
          class="tab-button"
          title="List of Nodes"
        >
          <button
            class="btn btn-circle btn-light"
          >
            <svg
              class="base-icon"
              height="100%"
              version="1.1"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <use
                x="0"
                xlink:href="../../assets/images/icons.svg#nodes"
                y="0"
              />
            </svg>
          </button>
        </div>
        <div
          class="tab-button"
          title="List of Edges"
        >
          <button
            class="btn btn-circle btn-light"
          >
            <svg
              class="base-icon"
              height="100%"
              version="1.1"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <use
                x="0"
                xlink:href="../../assets/images/icons.svg#edges"
                y="0"
              />
            </svg>
          </button>
        </div>
        <div
          class="tab-button"
          title="Raw Data"
        >
          <button
            class="btn btn-circle btn-light"
          >
            <svg
              class="base-icon"
              height="100%"
              version="1.1"
              width="100%"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <use
                x="0"
                xlink:href="../../assets/images/icons.svg#data"
                y="0"
              />
            </svg>
          </button>
        </div>
      </div>
       
      <transition-stub
        name="slide-x"
      >
        <!---->
      </transition-stub>
    </div>
  </transition-stub>
   
  <div
    class="view-container"
  >
    <div
      class="adb-view"
      draggable="true"
      id="adb-view"
      style="--move-x: 0px; --move-y: 0px; --scale: 1;"
      zoom="0"
    >
      <div
        class="view"
      >
        <div
          class="edge"
          style="--top_pos: 0px; --left_pos: 0px; --length: 0px; --angle: 0rad;"
        >
          <!---->
           
          <div
            class="arrow front"
          />
           
          <div
            class="element"
          >
            
    -1

          </div>
        </div>
         
        <div
          class="node"
          style="--node-size: 2.4rem; top: calc(0px - 1.2rem); left: calc(0px - 1.2rem);"
        >
          <div
            class="element"
          >
            
    1

          </div>
        </div>
        <div
          class="node"
          style="--node-size: 2.4rem; top: calc(0px - 1.2rem); left: calc(0px - 1.2rem);"
        >
          <div
            class="element"
          >
            
    2

          </div>
        </div>
      </div>
    </div>
     
    <div
      class="search-wrap"
    >
      <div
        class="search-field"
      >
        <transition-stub
          name="slide-x"
        >
          <!---->
        </transition-stub>
         
        <button
          class="btn btn-light"
        >
          <svg
            height="100%"
            width="100%"
          >
            <use
              x="0"
              xlink:href="../../assets/images/icons.svg#search"
              y="0"
            />
          </svg>
        </button>
      </div>
       
      <!---->
       
      <!---->
    </div>
  </div>
   
  <transition-stub
    name="fade"
  >
    <!---->
  </transition-stub>
   
  <transition-stub
    name="slide-x"
  >
    <div
      class="right-panel"
    >
      <div>
        <button
          class="btn btn-primary"
        >
          
                Hide all data
            
        </button>
         
        <transition-group-stub
          name="fade"
        >
          <div
            class="base-box"
          >
            <div
              class="header"
            >
              <div
                class="title"
              >
                <h4>
                  Node 1
                </h4>
              </div>
               
              <button
                class="btn btn-light toggle-button opened"
              >
                <svg
                  class="base-icon"
                  height="100%"
                  version="1.1"
                  width="100%"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                >
                  <use
                    x="0"
                    xlink:href="../../assets/images/icons.svg#arrow"
                    y="0"
                  />
                </svg>
              </button>
               
              <div
                class="controls"
              >
                <button
                  class="btn btn-light"
                >
                  <svg
                    height="100%"
                    width="100%"
                  >
                    <use
                      x="0"
                      xlink:href="../../assets/images/icons.svg#close"
                      y="0"
                    />
                  </svg>
                </button>
              </div>
            </div>
             
            <transition-stub
              name="slide-y"
            >
              <div
                class="content"
              >
                <table
                  class="element-data"
                >
                  <tr>
                    <td>
                      name
                    </td>
                     
                    <td>
                      Jana
                    </td>
                  </tr>
                  <tr>
                    <td>
                      number
                    </td>
                     
                    <td>
                      18
                    </td>
                  </tr>
                </table>
              </div>
            </transition-stub>
          </div>
          <div
            class="base-box"
          >
            <div
              class="header"
            >
              <div
                class="title"
              >
                <h4>
                  Edge -2
                </h4>
              </div>
               
              <button
                class="btn btn-light toggle-button opened"
              >
                <svg
                  class="base-icon"
                  height="100%"
                  version="1.1"
                  width="100%"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                >
                  <use
                    x="0"
                    xlink:href="../../assets/images/icons.svg#arrow"
                    y="0"
                  />
                </svg>
              </button>
               
              <div
                class="controls"
              >
                <button
                  class="btn btn-light"
                >
                  <svg
                    height="100%"
                    width="100%"
                  >
                    <use
                      x="0"
                      xlink:href="../../assets/images/icons.svg#close"
                      y="0"
                    />
                  </svg>
                </button>
              </div>
            </div>
             
            <transition-stub
              name="slide-y"
            >
              <div
                class="content"
              >
                <table
                  class="element-data"
                >
                  <tr>
                    <td>
                      foo
                    </td>
                     
                    <td>
                      bar
                    </td>
                  </tr>
                </table>
              </div>
            </transition-stub>
          </div>
        </transition-group-stub>
      </div>
    </div>
  </transition-stub>
   
  <div
    class="controls"
  >
    <button
      class="btn btn-default reset"
    >
      Reset view
    </button>
     
    <button
      class="btn btn-default plus"
    >
      +
    </button>
     
    <button
      class="btn btn-default minus"
    >
      -
    </button>
  </div>
</div>
`;
